pxr/usd/usdUtils/testenv/testUsdUtilsValidators.cpp

```diff
@@ -0,0 +1,102 @@
+#include "pxr/usd/usd/validator.h"
+#include "pxr/usd/usd/validationError.h"
+#include "pxr/usd/usdUtils/validatorTokens.h"
+#include "pxr/usd/usd/validationRegistry.h"
+#include "pxr/base/arch/systemInfo.h"
+
+PXR_NAMESPACE_USING_DIRECTIVE
+
+static
+void
+TestUsdUsdzValidators()
+{
+    // This should be updated with every new validator added with
+    // UsdUtilsValidators keyword.
+    UsdValidationRegistry& registry = UsdValidationRegistry::GetInstance();
+    UsdValidatorMetadataVector metadata =
+            registry.GetValidatorMetadataForKeyword(
+                    UsdUtilsValidatorKeywordTokens->UsdUtilsValidators);
+    // Since other validators can be registered with a UsdUtilsValidators
+    // keyword, our validators registered in usd are a subset of the entire
+    // set.
+    std::set<TfToken> validatorMetadataNameSet;
+    for (const UsdValidatorMetadata &metadata : metadata) {
+        validatorMetadataNameSet.insert(metadata.name);
+    }
+
+    const std::set<TfToken> expectedValidatorNames =
+            {UsdUtilsValidatorNameTokens->packageEncapsulationValidator};
+
+    TF_AXIOM(std::includes(validatorMetadataNameSet.begin(),
+                           validatorMetadataNameSet.end(),
+                           expectedValidatorNames.begin(),
+                           expectedValidatorNames.end()));
+}
+
+static
+void
+TestPackageEncapsulationValidator()
+{
+    UsdValidationRegistry& registry = UsdValidationRegistry::GetInstance();
+
+    // Verify the validator exists
+    const UsdValidator *validator = registry.GetOrLoadValidatorByName(
+            UsdUtilsValidatorNameTokens->packageEncapsulationValidator);
+
+    TF_AXIOM(validator);
+
+    // Load the pre-created usdz stage with paths to a layer and asset
+    // that are not included in the package, but exist
+    const UsdStageRefPtr& stage = UsdStage::Open(
+            "testUsdUtilsPackageEncapsulation/fail.usdz");
+
+    UsdValidationErrorVector errors = validator->Validate(stage);
+
+    // Verify both the layer & asset errors are present
+    TF_AXIOM(errors.size() == 2);
+
+    const std::string& realRootPath = stage->GetRootLayer()->GetRealPath();
+    const std::string& rootDirectory = realRootPath
+            .substr(0, realRootPath.find_last_of("/") + 1);
+
+    const std::string& errorLayer = rootDirectory + "excludedDirectory"
```

Comment:
And as mentioned ^, this does fail on windows because of this.

Reply:
I've updated this to use std::filesystem to build the paths properly, let me know if there's a better way. It seems when I just try to use the identifier of the root stage, it uses the relative path "pass.usdz" or "fail.usdz" but the error message from the code ends up including the full actual path in the error message, using the realPath of the root usdz file is the best way I can think of to get that root, however since it's known the file will be in the current working directory, maybe it makes sense to use something like `ArchGetCwd()` but I still think i need `std::filesystem` in order to append `excludedDirectory` and get the correct "/" or "\" afterwards.

let me know what you think

Reply:
Hi @beersandrew 

Do you think https://github.com/PixarAnimationStudios/OpenUSD/blob/dev/pxr/base/tf/pathUtils.h#L80-L88 might be helpful?

Reply:
appreciate your patience on this one @tallytalwar, I've updated the pathing to use Tf functions where it seemed appropriate, let me know what you think